### ISSUE TITLE
GAP Error shows up in Julia tests, but does not lead to failure (happens on Travis)

### ISSUE THREAD

**Issue by fingolfin** [oscar-system/GAP.jl#235]
The following is seen in a successful (!) [Travis build](https://travis-ci.com/oscar-system/GAP.jl/jobs/200649781)
```
...
JULIA_TRACK_COVERAGE enabled
Error, List Element: <list>[4] must have an assigned value
Test Summary: | Pass  Total
basics        |   15     15
...
```
We should fix this error; also I am kind of worried that it doesn't lead to a test failure; perhaps we should first improve our setup to ensure it does fail the Travis test, before actually fixing it.

**Comment by ThomasBreuer** [comment]
The error message is printed by GAP.
It comes from the input line `@test_throws ErrorException xx[4]` in the file `test/basics.jl` of GAP.jl,
where `xx` is a GAP list of length 3.
The intention is to test whether an error is signaled in this situation,
as suggested in the section "Unit Testing" of Julia's documentation.

Thus it is correct that the tests are regarded as successful.

The point seems to be to change the handling of GAP's error messages
such that, for example, the message is not printed in such a test situation.

**Comment by sebasguts** [comment]
Indeed. The error message there is intentional, and its appearance is right :)

**Comment by fingolfin** [comment]
Of course the error is intentional, but I disagree that its appearance in this place is right. Rather, this is a bug: GAP prints its error message directly to the terminal, which it should not. I am not sure *what* should happen, but not this.

**Comment by fingolfin** [comment]
I guess the underlying issue here is that ideally want to integrate the IO subsystems of GAP and Julia more closely, so that they don't interfere with each other as much. That's of course non-trivial work, and also it's not 100% clear what this means. Perhaps that GAP uses Julia IO functions (so ultimately: `libuv`) when built with Julia support?

**Comment by fingolfin** [comment]
This was resolved by PR #355
